### PR TITLE
icons.html: query the icons pages instead of the filesystem

### PR DESCRIPTION
**Pull request by XhmikosR**
This makes sure we only use any icons that have a content page. Also, we can use `.RelPermalink` instead of hardcoding the page path.

Draft because I notice linkinator scans -3 total links (3835 vs 3832) so something is different while it shouldn't. Maybe we are missing a page or something.

Requires #704

Preview: https://deploy-preview-729--bootstrap-icons.netlify.app/

### CODE REVIEW

**Comment by XhmikosR** [comment]
Found the missing links; it's the tags page that's being excluded for some reason.

```
[200] _site\icons\tags.svg
[200] _site\icons\tags\
```

Unsure why this is happening yet.

**Comment by XhmikosR** [comment]
So, apparently the previous solution was confusing Hugo. I switched to `.Site.GetPage (printf "icons/%s" $filename)` and everything seems to be fine. Which means that previously, the tags for the `tags` icon where missing :)

File: docs/layouts/partials/icons.html [hunk 1]
```diff
@@ -7,11 +7,11 @@ <h2 id="icons" class="mb-0">Icons</h2>
     </form>
   </div>
   <ul class="row row-cols-3 row-cols-sm-4 row-cols-lg-6 row-cols-xl-8 list-unstyled list">
-    {{ range (readDir "icons") -}}
-      {{- $filenameWithExt := split .Name "." -}}
-      {{- $filename := index $filenameWithExt 0 -}}
-      <li class="col mb-4"{{ with $.Site.GetPage "icons" $filename }}{{ with .Params.tags }} data-tags="{{ delimit . " " }}"{{ end }}{{ end }}>
-        <a class="d-block text-dark text-decoration-none" href="/icons/{{ $filename }}/">
+    {{ range (where site.RegularPages "Type" "icons") -}}
```
Comment (XhmikosR) [inline docs/layouts/partials/icons.html:10]: We might need to sort this by name. Marking the PR as draft until I fix this too

Comment (XhmikosR) [reply docs/layouts/partials/icons.html:10]: Might have something to do with the categories in the front matter. Categories should be an array.

Comment (XhmikosR) [reply docs/layouts/partials/icons.html]: Ah, found it! Those pages have the wrong title

Comment (XhmikosR) [reply docs/layouts/partials/icons.html]: I'll add a note to see if can make the partial fail in some cases like the above

File: docs/layouts/partials/icons.html [hunk 2]
```diff
@@ -7,11 +7,11 @@ <h2 id="icons" class="mb-0">Icons</h2>
...
   </div>
   <ul class="row row-cols-3 row-cols-sm-4 row-cols-lg-6 row-cols-xl-8 list-unstyled list">
-    {{ range (readDir "icons") -}}
-      {{- $filenameWithExt := split .Name "." -}}
-      {{- $filename := index $filenameWithExt 0 -}}
-      <li class="col mb-4"{{ with $.Site.GetPage "icons" $filename }}{{ with .Params.tags }} data-tags="{{ delimit . " " }}"{{ end }}{{ end }}>
-        <a class="d-block text-dark text-decoration-none" href="/icons/{{ $filename }}/">
+    {{ range (where .Site.RegularPages "Type" "icons") -}}
+      {{- $filename := .File.TranslationBaseName -}}
+      {{- with .Site.GetPage (printf "icons/%s" $filename) }}
```
Comment (XhmikosR) [inline docs/layouts/partials/icons.html:12]: Note to self, see if we really need this or if we can use another internal variable to get the page's path

**Comment by mdo** [comment]
Does this still fail our tests if we add a new SVG that doesn't have a page yet?

**Comment by XhmikosR** [comment]
From a quick thought without trying it, I don't think it will fail.

IMHO this is how we should have been doing it, since we require each icon to have its own content page, with its own title etc.

**Review by mdo — APPROVED** [review]

**Pull request merged**